sidorares/node-mysql2
# refactor: resolve small eslint rules and TS settings

About **1,209** additions and **1,791** deletions:
- Around **-904** is because `package-lock.json` by removing unused dependencies
- Around **+1,184** and **−735** it's just the `indentation: 2` from **ESLint** and `printWith: 80` from **Prettier**

> In practice, around **35** additions and **126** deletions were "*really*" made.

<hr>

### In this refacoring task:

> No types were changed in this **PR** except for `ban-types`
> See https://github.com/sidorares/node-mysql2/commit/ab0de35940f417db45dc80b28a39a5e78b25175a

- Resolve the following `eslint` rules:
  - indent
  - no-mixed-spaces-and-tabs
  - @typescript-eslint/no-unused-vars
  - @typescript-eslint/ban-types *

> Unfortunately, the `ban-types` resolution for `Connection` and `PoolConnection` will need to be resolved together with joining the spread types.

<hr>

- Remove unused **TS** tests
- Remove unused dependencies
  - `@types/mocha`, `@types/chai`, `mocha` and `chai` dependencies

<hr>

- Redefine the `tsconfig.json`
  - Include `index.d.ts`, `promise.d.ts` and the `typings` folder
  - Remove `"strictFunctionTypes": false`

<hr>

An easy way to review it, is looking for `commits`:
- https://github.com/sidorares/node-mysql2/commit/8f7445e31e71eb13700cd14c48ef35b6ba60657f (redefine TypeScript settings)
- https://github.com/sidorares/node-mysql2/commit/ae2f98f9032ed473a8801a77ea3cf63f2ca2317c (remove unused type tests)
- https://github.com/sidorares/node-mysql2/commit/ab0de35940f417db45dc80b28a39a5e78b25175a (resolve eslint ban-types)
- https://github.com/sidorares/node-mysql2/pull/2073/commits/744a49e0faa40523eb39eca754511f50cfde7a28 (rollback eslint ban-types for Connection and PoolConnection)
- etc.

And finally:

- https://github.com/sidorares/node-mysql2/commit/57770aa882e4d904e6e6924d42092c11304dd903 (resolve small eslint rules)
  -  `indentation` and `printWidth` 👻

<hr>

> See more in #2072
